decidim-core/app/commands/decidim/attachment_methods.rb

```diff
@@ -5,11 +5,14 @@ module Decidim
   module AttachmentMethods
     private
 
-    def build_attachment
+    def build_attachment(attached_to = nil)
+      attached_to = @attached_to if attached_to.blank?
+      attached_to = form.current_organization if attached_to.blank? && form.respond_to?(:current_organization)
+
       @attachment = Attachment.new(
         title: @form.attachment.title,
-        file: @form.attachment.file,
-        attached_to: @attached_to
+        attached_to: attached_to,
+        file: @form.attachment.file # Define attached_to before this
```

Comment:
Why do you need to define `attached_to` before `file`?

Reply:
Because we are getting the upload settings from the organization and before `attached_to` is set, there is no context information about to which organization the attachment (or any other record) belongs to.

When the file attribute gets set in this situation, it creates the attached uploader through Carrierwave. Then, if that uploader does not have the correct context information available, the organization specific settings would not apply and the defaults would be used instead.

Reply:
Another case is that if you want to have context specific settings (e.g. participant vs admin), this is also needed for the same reason. The context is fetched from the `attached_to` record in this case.

E.g. proposal attachments (with less file extensions) vs participatory space attachments (with more extensions).

Reply:
And just to emphasize what happens when you call `Attachment.new(attributes)`:
- The `attributes` hash is iterated in the same order as it is defined in.
- For each attribute in the `attributes` hash, the setter method for that attribute will be called for the new `Attachment` instance (e.g. `attachment.attached_to = attributes[:attached_to]`.
- When `attachment.file = attributes[:file]` is called, Carrierwave creates the uploader instance and runs the validations on the file object.
- This uploader instance needs the organization context in order to get the organization specific settings.
- If the organization has not been set before this, the organization specific settings are useless as the defaults will be used.

This is important because Carrierwave does some extra validation checks on the passed file (such as content type whitelist and extension whitelist) where the organization context matters. The form object also does the content type validation check but it is different from what Carrierwave does internally. The form object's validation happens through the [file_validators gem](https://github.com/musaffa/file_validators).